pennylane/transforms/dynamic_one_shot.py

```diff
@@ -58,40 +58,42 @@ def null_postprocessing(results):
 def dynamic_one_shot(tape: QuantumScript, **kwargs) -> tuple[QuantumScriptBatch, PostprocessingFn]:
     """Transform a QNode to into several one-shot tapes to support dynamic circuit execution.
 
+    This transform enables the ``"one-shot"`` mcm method. The ``"one-shot"`` method prompts the
+    device to perform a series of one-shot executions, where in each execution, the ``qml.measure``
+    operation dynamically measures the wire and collapses the state vector stochastically.
```

Comment:
This is information specific to state-vector simulators. `one-shot` only transforms the circuit, and it's up to the device to figure out how to support it. A device in hardware could use one-shot, in which case `dynamic_one_shot`'s job would be to post-process the results correctly.

Reply:
```suggestion
    This transform prepares the circuit for the ``"one-shot"`` mcm method and post-process the
    results returned from the device. The ``"one-shot"`` method prompts the device to perform a 
    series of one-shot executions, where in each execution, the ``qml.measure`` operation dynamically 
    measures the wire and collapses the state vector stochastically.
```
How about now? @mudit2812

Reply:
state vectors shouldn't be mentioned at all. Maybe the following:
```suggestion
    This transform enables the ``"one-shot"`` mid-circuit measurement method. The ``"one-shot"`` method prompts the
    device to perform a series of one-shot executions, where in each execution, the ``qml.measure``
    operation applies a probabilistic mid-circuit measurement to the circuit.
```